pandas/core/frame.py

```diff
@@ -611,7 +611,8 @@ def _is_homogeneous_type(self) -> bool:
         if self._mgr.any_extension_types:
             return len({block.dtype for block in self._mgr.blocks}) == 1
         else:
-            return not self._mgr.is_mixed_type
+            # Note: consolidates inplace
```

Comment:
i would rather just push this entire think to the manager

Reply:
semi-agree: we shouldnt mix-and-match.  But _item_cache pretty much has to be on DataFrame, since that is where the Series objects are created.